packages/cdk/lib/roles/toil-job-role.ts

```diff
@@ -0,0 +1,50 @@
+import { PolicyOptions } from "../types/engine-options";
+import { BucketOperations } from "../common/BucketOperations";
+import { Construct } from "constructs";
+import { Role, ServicePrincipal, PolicyDocument, PolicyStatement, Effect } from "aws-cdk-lib/aws-iam";
+
+export interface ToilJobRoleProps {
+  readOnlyBucketArns: string[];
+  readWriteBucketArns: string[];
+  policies: PolicyOptions;
+}
+
+// This role grants access to everything a Toil job needs to talk to the AWS
+// job store and/or additional user data in S3.
+export class ToilJobRole extends Role {
+  constructor(scope: Construct, id: string, props: ToilJobRoleProps, additionalInlinePolicies?: { [key: string]: PolicyDocument }) {
+    super(scope, id, {
+      assumedBy: new ServicePrincipal("ecs-tasks.amazonaws.com"),
+      inlinePolicies: {
+        // TODO: Remove this when Toil no longer uses its own SimpleDB domains
+        ToilSimpleDBFullAccess: new PolicyDocument({
+          assignSids: true,
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: ["sdb:*"],
+              resources: ["*"],
+            }),
+          ],
+        }),
+        // TODO: Remove this when Toil is taught to use AGC buckets to store
+        // its workflow state and doesn't need to make and destroy its own.
+        ToilS3FullAccess: new PolicyDocument({
+          assignSids: true,
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: ["s3:*"],
```

Comment:
This is far too permissive. It could be used to do literally anything to a customers buckets. This must be limited to a pattern or single bucket.

Reply:
Toil is going to make a buckets and a SimpleDB domains with names starting with `toil-cwl-` for each workflow, for scratch. It also I believe uses a `toil` SimpleDB domain.

We probably should be able to grant permissions for just those prefixes.

This will cause trouble if the user wants to specify their own `--jobStore` and tell Toil to use a different bucket, but we don't necessarily have to support that.